src/lib/eslint/BaseEslintEngine.ts

```diff
@@ -0,0 +1,228 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import {Logger, SfdxError, LoggerLevel} from '@salesforce/core';
+import {Catalog, Rule, RuleGroup, RuleResult, RuleTarget, RuleViolation, ESRule, ESReport, ESMessage} from '../../types';
+import {OutputProcessor} from '../pmd/OutputProcessor';
+import {RuleEngine} from '../services/RuleEngine';
+import {CLIEngine} from 'eslint';
+import * as path from 'path';
+
+
+export interface EslintStrategy {
```

Comment:
I'm not loving the name `strategy`, because it's not immediately clear what a strategy actually _means_. If we could either add some more commenting to describe what a strategy is, or change the name to something more descriptive, I feel that would be ideal. Incidentally, I'm always a fan of naming interfaces something that ends with "-able" or "-ible", or "-er" but that might not be realistic here.

Reply:
I'm using a fairly common design pattern called "Strategy pattern": https://en.wikipedia.org/wiki/Strategy_pattern
In general, when we use a design pattern, there's a recommendation to use names related to the pattern so that fellow-developers right away know what we are trying to do.